prow/scripts/upgrade-prow.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+set -o errexit
+
+readonly SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+# shellcheck disable=SC1090
+source "${SCRIPT_DIR}/library.sh"
+
+authenticate
+
+TAG=$(curl https://raw.githubusercontent.com/kubernetes/test-infra/master/prow/cluster/starter.yaml | grep image | grep deck | awk '{print $2}' | cut -d ':' -f2)
```

Comment:
What will happen if they add a new required parameter to an application? This script does not help us in such a situation.

Reply:
Have you checked how other projects approach that topic? Have you asked on a slack channel what are their recommendations?

Reply:
Yes, I have look into istio and knative, you can find my comment in #148 
Sadly, I don't think there is a good way of automating the deployment manifests. Istio and knaive both do it semi manually and update only the image tag.